File: G_Projects/I_Vending_Machine/VendingMachineStates/ConcreteStates/OutOfStockState.java
package G_Projects.I_Vending_Machine.VendingMachineStates.ConcreteStates;

import G_Projects.I_Vending_Machine.CommonEnums.Coin;
import G_Projects.I_Vending_Machine.VendingMachineStates.VendingMachineContext;
import G_Projects.I_Vending_Machine.VendingMachineStates.VendingMachineState;

public class OutOfStockState implements VendingMachineState {

    @Override
    public String getStateName() {
        return "Out Of Stock State";
    }

    @Override
    public void insertCoin(VendingMachineContext context, Coin coin) {
        throw new IllegalStateException("Machine is out of stock");
    }

    @Override
    public void selectItem(VendingMachineContext context, int codeNumber) {
        throw new IllegalStateException("Machine is out of stock");
    }

    @Override
    public void dispenseItem(VendingMachineContext context) {
        throw new IllegalStateException("Machine is out of stock");
    }
}

File: Python/A_Project/E_Vending_Machine/VendingMachineStates/ConcreteStates/out_of_stock_state.py
from VendingMachineStates.vending_machine_state import VendingMachineState


class OutOfStockState(VendingMachineState):
    def get_state_name(self) -> str:
        return "Out Of Stock State"

    def insert_coin(self, context, coin):
        raise RuntimeError("Machine is out of stock")

    def select_item(self, context, code_number: int):
        raise RuntimeError("Machine is out of stock")

    def dispense_item(self, context):
        raise RuntimeError("Machine is out of stock")
